src/Language/Haskell/Liquid/Bare/OfType.hs

```diff
@@ -148,6 +143,24 @@ ofBRType appRTAlias resolveReft
     go_syms
       = secondM ofBSort
 
+    goRFun bounds _ (RApp c ps _ _) t | Just bnd <- M.lookup c bounds 
+      = makeBound bnd (bar <$> ps) <$> go t
+    goRFun _ x t1 t2
+      = rFun x <$> go t1 <*> go t2
+
+    bar (RVar x _) = x
+    bar _          = errorstar "BAR"   
```

Comment:
+ @spinda is correct, can we please **not** use error/errorstar **ever**,
especially inside BareM, where `throwError` is there.

On Tue, Mar 10, 2015 at 2:24 PM, Michael Smith <notifications@github.com>
wrote:

> In src/Language/Haskell/Liquid/Bare/OfType.hs
> <https://github.com/ucsd-progsys/liquidhaskell/pull/358#discussion_r26167800>
> :
>
> > @@ -148,6 +143,24 @@ ofBRType appRTAlias resolveReft
> >      go_syms
> >        = secondM ofBSort
> >
> > +    goRFun bounds _ (RApp c ps _ _) t | Just bnd <- M.lookup c bounds
> > +      = makeBound bnd (bar <$> ps) <$> go t
> > +    goRFun _ x t1 t2
> > +      = rFun x <$> go t1 <*> go t2
> > +
> > +    bar (RVar x _) = x
> > +    bar _          = errorstar "BAR"
>
> errorstar is just a wrapper around error
> <https://github.com/ucsd-progsys/liquid-fixpoint/blob/b1592e1d8189a8f8ba5c2920e0c8eaffdc9e607d/src/Language/Fixpoint/Misc.hs#L88>
> .
>
> ofBRType is already in BareM, so we should throw a proper Error via
> throwError instead of using errorstar here.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell/pull/358/files#r26167800>.
>



-- 
Ranjit.